File: Models/Member.cs
using System.ComponentModel.DataAnnotations;

namespace BookPulse.Models
{
    public class Member
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string LastName { get; set; } = string.Empty;

        [Required]
        [EmailAddress]
        [StringLength(200)]
        public string Email { get; set; } = string.Empty;

        [StringLength(20)]
        public string? PhoneNumber { get; set; }

        public DateTime RegistrationDate { get; set; } = DateTime.UtcNow;

        public bool IsActive { get; set; } = true;

        // Navigation properties
        public virtual ICollection<Loan> Loans { get; set; } = new List<Loan>();
        public virtual ICollection<Rating> Ratings { get; set; } = new List<Rating>();
    }
}

File: frontend/src/types/member.ts
export interface Member {
  id: number
  firstName: string
  lastName: string
  email: string
  phoneNumber?: string
  registrationDate: string
  isActive: boolean
}

export interface CreateMemberDto {
  firstName: string
  lastName: string
  email: string
  phoneNumber?: string
}

export interface UpdateMemberDto {
  firstName?: string
  lastName?: string
  email?: string
  phoneNumber?: string
  isActive?: boolean
}
